hack/update-bazel.sh

```diff
@@ -20,6 +20,9 @@ set -o pipefail
 TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
 TMP_GOPATH=$(mktemp -d)
 
+# no unit tests in vendor
+find ${TESTINFRA_ROOT}/vendor/ -name "*_test.go" -delete
```

Comment:
I kinda feel like putting this in the update-bazel script is misplaced. Should we have a `go dep` wrapper instead?

Reply:
it's not exactly the clearest, but because updating bazel is required anytime you run dep (it blows away all `BUILD` files in vendor/), it's not the worst place either.

Reply:
Maybe? `hack/update-bazel.sh` seemed like our "run after changing the build" which also fit "run after changing vendor/"

Reply:
We can't really enforce that anyone uses our dep wrapper as well, so we'd still need to have a `verify` somewhere.

Reply:
Oh hmm @cblecker your comment hadn't loaded (??), that sums up what I was thinking more concisely 🙃

Reply:
it works for me, as the current workflow is always
`dep ensure`
`dep prune`
`./hack/update-bazel.sh`

Or maybe we can also make use of the `hack/update-all.sh` if it should not be in `update-bazel.sh`?